src/main/java/me/botsko/prism/database/mysql/MySQLPrismDataSource.java

```diff
@@ -54,10 +74,46 @@ public void setFile() {
 
     @Override
     public SelectQuery createSelectQuery() {
-        if (nonStandardSQL) {
+        if (nonStandardSql) {
             return new MySQLSelectQueryBuilder(this);
         } else {
             return new SQLSelectQueryBuilder(this);
         }
     }
+
+    private void detectNonStandardSql() {
+        try (
+                PreparedStatement st = getConnection().prepareStatement("SHOW VARIABLES");
+                ResultSet rs = st.executeQuery()) {
+            while (rs.next()) {
+                dbInfo.put(rs.getString(1).toLowerCase(), rs.getString(2));
+            }
+        } catch (SQLException e) {
+            Prism.debug(e.getMessage());
+        }
+        boolean anyValueSuccess;
+        try (
+                PreparedStatement st = getConnection().prepareStatement("SELECT ANY_VALUE(1)");
+                ResultSet rs = st.executeQuery()) {
+            nonStandardSql = true;
+            anyValueSuccess = true;
+            rs.next();
+        } catch (SQLException e) {
+            anyValueSuccess = false;
+        }
+        String version = dbInfo.get("version");
+        String versionComment = dbInfo.get("version_comment");
+        Prism.log("Prism detected you database is " + version + " / " + versionComment);
+        Prism.log("You have set nonStandardSql to " + nonStandardSql);
+        if (!anyValueSuccess) {
+            if (nonStandardSql) {
```

Comment:
As per Codacy suggestion, I was also thinking you could combine these instead of nesting?
```
if ((!anyValueSuccess) && (nonStandardSql)) {
```